packages/core-contracts/contracts/token/ERC20-EIP2612.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.8.0;
+
+import "./ERC20.sol";
+
+/// @notice Modern and gas efficient ERC20 + EIP-2612 implementation.
+/// @author Modified from Uniswap (https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2ERC20.sol)
+/// @author Modified from Rari-Capital (https://github.com/Rari-Capital/solmate/blob/main/src/erc20/ERC20.sol)
+abstract contract ERC20_2612 is ERC20 {
+    bytes32 public constant PERMIT_TYPEHASH =
```

Comment:
Should we pre-calculate this hash and leave the source of the calculation as a comment to save gas?

Reply:
You can use bytes32 literals in Soldity.
E.g: https://github.com/Synthetixio/synthetix-v3/blob/main/packages/deployer/test/fixture-projects/sample-project/contracts/storage/OwnerStorage.sol#L13